Readme.md

```diff
@@ -27,10 +27,35 @@ MiniTrainer is a small form factor and extremely efficient training library for
 
 # Installation
 
+## Using uv (Recommended)
+
 ```shell
-conda create -n minitrain python=3.12 -y; conda activate minitrain
-pip install torch
-pip install -r requirements.txt
+# Regular installation
+uv sync
+
+# Install with flash-attention support (flash-attn installed separately due to build dependencies)
+uv sync
+uv pip install .[flash]
+
+# For editing this codebase, install with -e (editable)
+uv pip install -e .
+uv pip install -e .[flash]
+```
+
+## Using pip
+
+```shell
+# Create virtual environment
+python -m venv .venv
+source .venv/bin/activate
+
+# Regular installation
+pip install .
+pip install .[flash]
```

Comment:
Only comment: should the extra be called "cuda" to maintain parity with other libraries? Also, that way if we need to add other cuda/gpu specific packages down the line, they can be directly added, instead of renaming down the line

Reply:
Sure, that would make it easy to be consistent.